src/pages/AMM/Pools/components/DepositForm/DepositForm.tsx

```diff
@@ -1,131 +1,168 @@
-import { zodResolver } from '@hookform/resolvers/zod';
-import { LiquidityPool, newMonetaryAmount } from '@interlay/interbtc-api';
+import { CurrencyExt, LiquidityPool, newMonetaryAmount, PooledCurrencies } from '@interlay/interbtc-api';
+import { AccountId } from '@polkadot/types/interfaces';
 import Big from 'big.js';
-import { useForm } from 'react-hook-form';
+import { ChangeEventHandler, FormEventHandler, RefObject, useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { useMutation } from 'react-query';
 import { toast } from 'react-toastify';
-import * as z from 'zod';
 
 import { displayMonetaryAmount, displayMonetaryAmountInUSDFormat } from '@/common/utils/utils';
-import { Dd, Dl, DlGroup, Dt, Flex, P, TokenInput } from '@/component-library';
+import { Dd, DlGroup, Dt, Flex, TokenInput } from '@/component-library';
 import { AuthCTA } from '@/components/AuthCTA';
-import { GOVERNANCE_TOKEN, TRANSACTION_FEE_AMOUNT } from '@/config/relay-chains';
-import validate, { PoolDepositSchemaParams } from '@/lib/form-validation';
-import { getErrorMessage, isValidForm } from '@/utils/helpers/forms';
+import { TRANSACTION_FEE_AMOUNT } from '@/config/relay-chains';
+import { SlippageManager } from '@/pages/AMM/shared/components';
+import { AMM_DEADLINE_INTERVAL } from '@/utils/constants/api';
 import { getTokenPrice } from '@/utils/helpers/prices';
 import { useGetBalances } from '@/utils/hooks/api/tokens/use-get-balances';
 import { useGetPrices } from '@/utils/hooks/api/use-get-prices';
+import useAccountId from '@/utils/hooks/use-account-id';
 
 import { PoolName } from '../PoolName';
+import { DepositDivider } from './DepositDivider';
 import { StyledDl } from './DepositForm.styles';
+import { DepositOutputAssets } from './DepositOutputAssets';
+import { useFormState } from './use-form-state';
 
-type DepositFormData = Record<string, string>;
+type DepositData = {
+  amounts: PooledCurrencies;
+  pool: LiquidityPool;
+  slippage: number;
+  deadline: number;
+  accountId: AccountId;
+};
+
+const mutateDeposit = ({ amounts, pool, slippage, deadline, accountId }: DepositData) =>
+  window.bridge.amm.addLiquidity(amounts, pool, slippage, deadline, accountId);
 
 type DepositFormProps = {
   pool: LiquidityPool;
-  onChangePool?: () => void;
+  slippageModalRef: RefObject<HTMLDivElement>;
+  onDeposit?: () => void;
 };
 
-const DepositForm = ({ pool }: DepositFormProps): JSX.Element => {
+const DepositForm = ({ pool, slippageModalRef, onDeposit }: DepositFormProps): JSX.Element => {
+  const accountId = useAccountId();
```

Comment:
Could this be tidied up a bit? My preferred order would be variables first (where those variables are being calculated from a prop and used to set default state), then instances of `useState`, and then calls to hooks. I think that would look something like this:

```
const { pooledCurrencies } = pool;
const defaultValues = pooledCurrencies.reduce((acc, amount) => ({ ...acc, [amount.currency.ticker]: undefined }), {});

const [slippage, setSlippage] = useState(0.1);
const [values, setValues] = useState<Record<string, number | undefined>>(defaultValues);

const accountId = useAccountId();
const { t } = useTranslation();
const { getAvailableBalance } = useGetBalances();
const prices = useGetPrices();

etc. etc.
```